### PR TITLE
Integrate Dashboard View with Widgets

### PR DESCRIPTION
**Pull request by jhoelscher7**
- Moved dashboard loading code to dashboard view component
- Linked dashboard list elements to their respective dashboard
- Modified layout to use bootstrap
- Chart Improvements and fixes
- Bug fix when switching between loaded dashboards
- Login Intercept

### CODE REVIEW

File: src/app/widget_modules/build/build-widget/build-widget.component.spec.ts [hunk 1]
```diff
@@ -10,7 +10,6 @@ import { DashboardService } from 'src/app/shared/dashboard.service';
 import { SharedModule } from 'src/app/shared/shared.module';
 
 import { GET_DASHBOARD_MOCK, POST_DASHBOARD_MOCK } from '../../../shared/dashboard.service.mockdata';
-import { BuildConfigFormComponent } from '../build-config-form/build-config-form.component';
 import { BuildService } from '../build.service';
 import { Build } from '../interfaces';
```
Comment (Sbrenthughes) [inline src/app/widget_modules/build/build-widget/build-widget.component.spec.ts]: Should this be renamed to IBuild?

File: src/app/shared/templates/base-template/base-template.component.ts [hunk 2]
```diff
@@ -1,7 +1,8 @@
-import { Component, ComponentFactoryResolver, ChangeDetectorRef, QueryList } from '@angular/core';
+import { ChangeDetectorRef, Component, ComponentFactoryResolver, QueryList } from '@angular/core';
+
 import { IWidget } from '../../interfaces';
-import { WidgetDirective } from '../../widget/widget.directive';
 import { WidgetComponent } from '../../widget/widget.component';
+import { WidgetDirective } from '../../widget/widget.directive';
```
[the remark this answers is not in the document]
Comment (jhoelscher7) [reply src/app/shared/templates/base-template/base-template.component.ts]: The imports should be sorted first by type (external, internal), then alphabetically.

File: src/app/screen_modules/team-dashboard/capone-template/capone-template.component.html [hunk 3]
```diff
@@ -24,3 +25,73 @@
     </div>
   </div>
 </div>
+-->
+
+<!-- <div class="container">
```
[the remark this answers is not in the document]
Comment (jhoelscher7) [reply src/app/screen_modules/team-dashboard/capone-template/capone-template.component.html]: I have swapped the flexbox layout for bootstrap for now. Testing out different layouts. For now I will remove all the commented code except the original flexbox layout nireesh included. which will stay there but commented out.

File: src/app/screen_modules/team-dashboard/dashboard-view/dashboard-view.component.ts [hunk 4]
```diff
@@ -12,13 +16,25 @@ import { BuildWidgetComponent } from 'src/app/widget_modules/build/build-widget/
...
+              private route: ActivatedRoute,
+              private dashboardService: DashboardService) {
     super(componentFactoryResolver, cdr);
   }
 
   ngOnInit() {
+    this.dashboardService.clearDashboard();
+    console.log(this.route.snapshot);
+    this.dashboardId = this.route.snapshot.paramMap.get('id');
+    console.log(this.dashboardId);
```
[the remark this answers is not in the document]
Comment (jhoelscher7) [reply src/app/screen_modules/team-dashboard/dashboard-view/dashboard-view.component.ts]: Fixing

File: src/app/shared/dashboard.service.ts [hunk 5]
```diff
@@ -13,17 +13,26 @@ export class DashboardService {
...
+  private dashboardId: string;
+
+  public dashboardConfig$ = this.dashboardSubject.asObservable().pipe(filter(result => result));
 
   constructor(private http: HttpClient) { }
 
   // Retrieve a new dashboard from the API, and push it to subscribers
   loadDashboard(dashboardId: string) {
+    this.dashboardId = dashboardId;
+    console.log(this.dashboardId);
```
[the remark this answers is not in the document]
Comment (jhoelscher7) [reply src/app/shared/dashboard.service.ts]: Fixing

File: src/app/widget_modules/build/build-widget/build-widget.component.spec.ts (hunk 1 above)
Comment (jhoelscher7) [reply src/app/widget_modules/build/build-widget/build-widget.component.spec.ts]: Yes I will rename all the interfaces for build.

File: src/app/shared/shared.module.ts [hunk 6]
```diff
@@ -14,28 +19,26 @@ import { ComboSeriesVerticalComponent } from './charts/combo-series-vertical/com
...
 import { WidgetDirective } from './widget/widget.directive';
-import { BaseTemplateComponent } from './templates/base-template/base-template.component';
-import {BuildWidgetComponent} from '../widget_modules/build/build-widget/build-widget.component';
-import {FlexLayoutModule} from '@angular/flex-layout';
-import {DragDropModule} from '@angular/cdk/drag-drop';
-import {CaponeTemplateComponent} from '../screen_modules/team-dashboard/capone-template/capone-template.component';
 
 @NgModule({
   declarations: [WidgetComponent, TwoByTwoLayoutComponent, LayoutComponent, ChartDirective,
     ChartComponent, LayoutDirective, LineChartComponent, NumberCardChartComponent,
```
[the remark this answers is not in the document]
Comment (jhoelscher7) [reply src/app/shared/shared.module.ts]: Cleaned up.

File: src/app/screen_modules/team-dashboard/dashboard-view/dashboard-view.component.ts [hunk 7]
```diff
@@ -12,23 +16,32 @@ import { BuildWidgetComponent } from 'src/app/widget_modules/build/build-widget/
...
     super(componentFactoryResolver, cdr);
   }
 
   ngOnInit() {
+    this.dashboardService.clearDashboard();
+    this.dashboardId = this.route.snapshot.paramMap.get('id');
+    this.dashboardService.loadDashboard(this.dashboardId);
+
     // get template type for dashboard and create template component
     // noinspection TypeScriptValidateTypes
```
[the remark this answers is not in the document]
Comment (jhoelscher7) [reply src/app/screen_modules/team-dashboard/dashboard-view/dashboard-view.component.ts]: Cleaned up.

**Pull request merged**